service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -87,14 +139,29 @@ private void insertAssessmentResult(
     claimRepository.save(claimEntity);
   }
 
+  @Override
+  public void setOffRampReason(Claim claimWithOffRamp) {
+    Optional<ClaimSubmissionEntity> claimSubmission =
+        claimSubmissionRepository.findFirstByReferenceIdAndIdTypeOrderByCreatedAtDesc(
+            String.valueOf(claimWithOffRamp.getCollectionId()), Claim.DEFAULT_ID_TYPE);
+    ClaimSubmissionEntity claimSubmissionEntity = claimSubmission.get();
+    claimSubmissionEntity.setOffRampReason(claimWithOffRamp.getOffRampReason());
+    claimSubmissionRepository.save(claimSubmissionEntity);
+  }
+
   @Override
   public void insertEvidenceSummaryDocument(GeneratePdfPayload request, String documentName) {
-    ClaimEntity claim =
-        claimRepository.findByClaimSubmissionId(request.getClaimSubmissionId()).orElse(null);
-    if (claim == null) {
-      log.warn("Could not find claim by claimSubmissionId, exiting.");
+    // ClaimSubmissionId in v1 calls is equal to ReferenceId on ClaimSubmission in VRO
+    Optional<ClaimSubmissionEntity> claimSubmission =
+        claimSubmissionRepository.findFirstByReferenceIdAndIdTypeOrderByCreatedAtDesc(
+            request.getClaimSubmissionId(), Claim.DEFAULT_ID_TYPE);
```

Comment:
This would be a bug for v2 endpoints.
```suggestion
            request.getClaimSubmissionId(), claim. getIdType());
```